xbmc/windowing/android/AndroidUtils.cpp

```diff
@@ -348,3 +370,39 @@ void CAndroidUtils::OnSettingChanged(const std::shared_ptr<const CSetting>& sett
   if (settingId == CAndroidUtils::SETTING_LIMITGUI)
     CDisplaySettings::GetInstance().ClearCalibrations();
 }
+
+std::vector<int> CAndroidUtils::GetDisplaySupportedHdrTypes() const
+{
+  CJNIWindow window = CXBMCApp::getWindow();
+
+  if (window)
+  {
+    CJNIView view = window.getDecorView();
+    if (view)
+    {
+      CJNIDisplay display = view.getDisplay();
+      if (display)
+      {
+        CJNIDisplayHdrCapabilities caps = display.getHdrCapabilities();
+        if (caps)
+          return caps.getSupportedHdrTypes();
+      }
+    }
+  }
+
+  return {};
+}
+
+void CAndroidUtils::LogDisplaySupportedHdrTypes() const
+{
+  const std::vector<int> hdrTypes = GetDisplaySupportedHdrTypes();
+  std::string text;
+
+  for (int type : hdrTypes)
```

Comment:
```suggestion
  for (const auto& type : hdrTypes)
```

Reply:
what's wrong with int here? Are we going to replace all types, even simplest ones with `auto`? ;-)

Reply:
`const int&`  then, but I prefer auto in such cases :)

Reply:
i agree with @ksooo "auto" is not really needed i could understand if it was an iterator or some else of long...

Reply:
![Captura de pantalla 2021-10-31 201030](https://user-images.githubusercontent.com/58434170/139597951-d4d13568-9009-4811-a36e-9b74cec3cf6e.png)

Reply:
> const int& then, but I prefer auto in such cases :)

 As you say it's all about personal preference, there is no right and wrong here. And we don't have a coding guideline for this, iirc. I alos use const auto& heavily, but not for trivial types and never if the code gets longer due to this. This contradicts the idea of auto.
 
  Let the submitter of this PR decide on this.

Reply:
Seems use `const &` is more efficient:
https://github.com/xbmc/xbmc/pull/18667

I said it for this, rather than for auto/int

Reply:
Will change to `const int&`.